realm-jni/src/util.cpp

```diff
@@ -278,6 +278,10 @@ string concat_stringdata(const char *message, StringData strData)
 
 jstring to_jstring(JNIEnv* env, StringData str)
 {
+    if (str.is_null()) {
+        return NULL;
```

Comment:
`to_string()` is an aux. function and not a method. It does not have access to `m_is_null`.